Zotero/Scenes/Detail/DetailCoordinator.swift

```diff
@@ -912,23 +924,60 @@ extension DetailCoordinator: DetailNoteEditorCoordinatorDelegate {
+            presenter.popViewController(animated: true)
         }
         let controller = UIHostingController(rootView: view.environmentObject(viewModel))
         controller.preferredContentSize = CGSize(width: SingleCitationViewController.width, height: CGFloat(values.count * 44))
-        self.citationNavigationController?.preferredContentSize = controller.preferredContentSize
-        self.citationNavigationController?.pushViewController(controller, animated: true)
+        presenter.preferredContentSize = controller.preferredContentSize
+        presenter.pushViewController(controller, animated: true)
     }
 
-    func showCitationPreview(errorMessage: String) {
+    func showCitationPreviewError(using presenter: UINavigationController, errorMessage: String) {
         let controller = UIAlertController(title: L10n.error, message: errorMessage, preferredStyle: .alert)
         controller.addAction(UIAlertAction(title: L10n.ok, style: .cancel, handler: nil))
-        self.citationNavigationController?.present(controller, animated: true, completion: nil)
+        presenter.present(controller, animated: true, completion: nil)
+    }
+
+    func showMissingStyleError(using presenter: UINavigationController?) {
+        guard let resolvedPresenter = presenter ?? navigationController else { return }
+        let controller = UIAlertController(title: L10n.error, message: L10n.Errors.Citation.missingStyle, preferredStyle: .alert)
+        controller.addAction(UIAlertAction(title: L10n.cancel, style: .cancel, handler: nil))
+        controller.addAction(UIAlertAction(title: L10n.Errors.Citation.openSettings, style: .default, handler: { _ in
+            openExportSettings(using: resolvedPresenter)
+        }))
+
+        if resolvedPresenter.presentedViewController == nil {
+            resolvedPresenter.present(controller, animated: true)
+        } else {
+            resolvedPresenter.dismiss(animated: true) {
+                resolvedPresenter.present(controller, animated: true)
+            }
+        }
+
+        func openExportSettings(using presenter: UINavigationController) {
+            let navigationController = NavigationViewController()
+            let containerController = ContainerViewController(rootViewController: navigationController)
+            let coordinator = SettingsCoordinator(startsWithExport: true, navigationController: navigationController, controllers: controllers)
+            coordinator.parentCoordinator = self
+            childCoordinators.append(coordinator)
+            coordinator.start(animated: false)
+            presenter.present(containerController, animated: true)
+        }
+    }
+}
+
+extension DetailCoordinator: DetailCopyBibliographyCoordinatorDelegate {
+    func overlay(for viewController: CopyBibliographyViewController, isHidden: Bool) {
+        guard isHidden else { return }
+        viewController.willMove(toParent: nil)
```

Comment:
I'd move this to `CopyBibliographyViewController`. The controller removing itself from parent controller is similar to `popViewController()` or `dismiss` which we call from the controller too. Also having the `isHidden` flag and ignoring half of it is a bit unnecessary.

Reply:
Initially I did have it as part of `CopyBibliographyViewController`, but it doesn't seem the right place semantically. We cannot assume that the view controller knows that it is is presented in a container, so that it would remove itself as such. That's why I added the logic for dismissal, in the same level that handled the initial presentation. 
The cases you refer to only work when the circumstances exist, which admittedly are of the most common ones. E.g. for `dismiss` we assume there is a presenting view controller to handle the dismissal.

Reply:
When hiding any view controller we always assume how it's shown and hide it based on that assumption. It works because all view controllers are (almost) always shown the same way (either in navigation controller or as modals). This view controller is pretty much only an overlay, so you can safely assume this won't even be pushed to a navigation controller or it won't be presented as a modal.

This also brings up a question, why isn't this presented as a modal over current context?

Reply:
I refactored this from a number of views that were part of `ItemsViewController`, so it is used in a similar manner to maintain the same functionality (check https://github.com/zotero/zotero-ios/pull/827/commits/196133df7240a06446d437de13bc44797621b4e1). We can improve it afterwards of course.

Reply:
Yes, I implemented it as a simple subview overlay before, because it was only used there. But since it's extracted into a view controller, which can be shown in multiple view controllers, I think you can safely keep everything the same as it's now and just present it with `modalPresentationStyle = .overCurrentContext`?

Reply:
Check https://github.com/zotero/zotero-ios/pull/827/commits/7b4c1a6a242b77c079e1ac995801d4ab7e30008f